File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.ts
function maxOperations(nums: number[], k: number): number {
    let operation:number = 0;
    let pairs:object = {};
    for(let num of nums)
    {
        if (pairs[k - num] > 0)
        {
            operation++;
            pairs[k - num]--;
        }
        else 
            pairs[num] = (pairs[num] | 0) + 1;
    }
    return operation;
};

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int result = 0,left = 0,right= nums.size()-1;
        sort(nums.begin(),nums.end());
        while(left < right)
        {
            if (nums[right] + nums[left] == k)
            {
                right--,left++,result++;
            }
            else if (nums[right] + nums[left] > k) right--;
            else left++;
        }
        return result;
    }
};